src/borg/archiver/prune_cmd.py

```diff
@@ -215,7 +215,11 @@ def build_parser_prune(self, subparsers, common_parser, mid_common_parser):
         The prune command prunes a repository by deleting all archives not matching
         any of the specified retention options.
 
-        Important: Repository disk space is **not** freed until you run ``borg compact``.
+        Important:
+
+        - Repository disk space is **not** freed until you run ``borg compact``.
```

Comment:
```suggestion
        - The prune command will only mark archives for deletion
          ("soft-deleted archives"), repository disk space is **not**
          freed until you run ``borg compact``.
```

Reply:
From a user perspective, this phrasing might be confusing. Sounds like there was some tag "delete-this" or some other visible mark, but the archive still being normally visible.

But from a user perspective it is gone (except when used special stuff like the ``--deleted`` flag or a undelete dry-run).

Reply:
Good point. Then let's explain it better. How about the following:

```
**Important:** Borg doesn't delete archives right away with the prune and
delete commands, but rather internally marks them for deletion. Borg generally
assumes these "soft-deleted archives" to not exist and thus also doesn't show
them with the list command (unless the ``--deleted`` option is given). However,
"soft-deleted archives" still take up disk space and can be restored with the
undelete command. To free the disk space, run ``borg compact``; this will
actually remove the archive and all its data from the repository. Consequently,
``borg undelete`` can't undelete archives after ``borg compact`` has been
executed.
```

The exact same explanation (I intentionally mentioned both `delete` and `prune` to emphasize that they work the same) can be used for `delete`.

It's quite verbose, I thus also thought about moving this to `compact`. But I feel like it's more important to explain this here.

Reply:
Solved differently. See undelete/compact help.